### PR TITLE
Move cache_dir config option to ForkConfig - feat/multichain

### LINKED ISSUE #740: Move `cache_dir` config option to `ForkConfig`
**Definition of Done**
- The `cache_dir` member field of `ProviderConfig` should be moved to `ForkConfig`
- Do this in the `feat/multichain` and `feat/solidity-tests` branches

### PR DESCRIPTION
**Pull request by iosh**
resolve #740

### CODE REVIEW

File: crates/edr_napi/src/config.rs [hunk 1]
```diff
@@ -256,7 +256,7 @@ impl TryFrom<ProviderConfig> for edr_napi_core::provider::Config {
             bail_on_call_failure: value.bail_on_call_failure,
             bail_on_transaction_failure: value.bail_on_transaction_failure,
             block_gas_limit,
-            cache_dir: value.cache_dir,
```
Comment (Wodann) [inline crates/edr_napi/src/config.rs:259]: The `cache_dir` should be moved on the `edr_provider::ProviderConfig` too

**Review by Wodann — CHANGES_REQUESTED** [review]

**Comment by iosh** [comment]
Please review this PR first. If everything is fine, I will update the other one.

File: crates/edr_napi/Cargo.toml [hunk 2]
```diff
@@ -33,6 +33,7 @@ tracing-flame = { version = "0.2.0", default-features = false, features = ["smal
 tracing-subscriber = { version = "0.3.18", default-features = false, features = ["ansi", "env-filter", "fmt", "parking_lot", "smallvec", "std"] }
 parking_lot = { version = "0.12.1", default-features = false }
 lazy_static = { version = "1.4.0", features = [] }
+edr_defaults = { version = "0.3.5", path = "../edr_defaults" }
```
Comment (Wodann) [inline crates/edr_napi/Cargo.toml:36]: The version can be omitted.

Please also try to adhere to alphabetical ordering

**Review by Wodann — CHANGES_REQUESTED** [review]
Thanks for the changes. Almost there!

One more request and then I'll validate CI

**Pull request closed without merging**